### ISSUE TITLE
sslv3 alert handshake failure

### ISSUE THREAD

**Issue by zzq889** [KrauseFx/fastlane#395]
It seems that Apple dropped SSLv3 support.
```
[11:24:56]: SSL_connect returned=1 errno=0 state=SSLv2/v3 read server hello A: sslv3 alert handshake failure (OpenSSL::SSL::SSLError)
[11:25:00]: fastlane finished with errors
/usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/ssl_socket.rb:120:in `connect_nonblock': SSL_connect returned=1 errno=0 state=SSLv2/v3 read server hello A: sslv3 alert
handshake failure (OpenSSL::SSL::SSLError) (Excon::Errors::SocketError)
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/ssl_socket.rb:120:in `initialize'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/connection.rb:387:in `new'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/connection.rb:387:in `socket'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/connection.rb:106:in `request_call'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/middlewares/mock.rb:47:in `request_call'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/middlewares/instrumentor.rb:22:in `request_call'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/middlewares/base.rb:15:in `request_call'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/middlewares/base.rb:15:in `request_call'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/middlewares/base.rb:15:in `request_call'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon/connection.rb:233:in `request'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/excon-0.45.3/lib/excon.rb:229:in `get'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/itunes_connect/itunes_connect_submission.rb:27:in `block in put_build_into_beta_testing!'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/itunes_connect/itunes_connect_submission.rb:26:in `loop'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/itunes_connect/itunes_connect_submission.rb:26:in `put_build_into_beta_testing!'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/ipa_uploader.rb:113:in `publish_beta_build'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/ipa_uploader.rb:93:in `publish_on_itunes_connect'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/ipa_uploader.rb:77:in `upload!'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/deliver_process.rb:384:in `trigger_ipa_upload'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/deliver_process.rb:48:in `run'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/deliverer.rb:133:in `finished_executing_deliver_file'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/deliverfile/deliverfile.rb:31:in `initialize'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/deliverer.rb:85:in `new'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/deliver-0.12.1/lib/deliver/deliverer.rb:85:in `initialize'
        from /usr/local/opt/rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/fastlane-1.7.0/lib/fastlane/actions/deliver.rb:21:in `new'
```

**Comment by KrauseFx** [comment]
Probably related to https://github.com/fastlane/spaceship/issues/54.

This might have something to do with your OpenSSL version.

**Comment by zzq889** [comment]
I follow the link to modified the spaceship client.rb to force openssl use TLSv1.2 but I still got the same error when I tried to deploy the app to the testflight. The build has already uploaded to the iTunes connect, but it get stucked at `Choosing the latest build on iTunesConnect for beta distribution` with the sslv3 alert failure. I use openssl(0.9.8zf) under OS X El Capitan.

**Comment by zzq889** [comment]
I think if it is possible to enforce `excon`'s ssl_socket to use TLS to replace SSL?

**Comment by zzq889** [comment]
Probably related to [excon/excon#478](https://github.com/excon/excon/issues/478)

**Comment by zzq889** [comment]
problem solved with version 1.13.0

**Comment by KrauseFx** [comment]
Great to hear :+1:
